spring-kafka/src/test/java/org/springframework/kafka/support/micrometer/ObservationTests.java

```diff
@@ -511,6 +520,20 @@ public void onSuccess(ProducerRecord<Integer, String> producerRecord, RecordMeta
 		tracer.getSpans().clear();
 	}
 
+	@Test
+	void testReplyingKafkaTemplateObservation(
+			@Autowired ReplyingKafkaTemplate<Integer, String, String> template,
+			@Autowired ObservationRegistry observationRegistry) {
+		AtomicReference<KafkaRecordReceiverContext> replyObservationContext = new AtomicReference<>();
+		template.sendAndReceive(new ProducerRecord<>(OBSERVATION_TEST_4, "test")).thenAccept(replyRecord -> {
+			Observation.Context observationContext = observationRegistry.getCurrentObservation().getContext();
+			assertThat(observationContext).isInstanceOf(KafkaRecordReceiverContext.class);
+			replyObservationContext.set((KafkaRecordReceiverContext) observationContext);
+		});
+		Awaitility.await().atMost(Duration.ofSeconds(60)).until(() ->
```

Comment:
See AssertJ API for `CompletableFuture` instead.
That `template.sendAndReceive()` returns for us exactly one we would like to verify.

Reply:
Much better, fixed.